### PR TITLE
feat: validate feed_info.feed_lang and agency.agency_lang match

### LINKED ISSUE #524: [REQUEST] Mismatching feed and agency language fields
**Describe the solution you'd like**
Validate feed and agency language fields match

**Additional context**
`v1.4.0` implementation:
- https://github.com/MobilityData/gtfs-validator/blob/v1.4.0/usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateAgencyLangAndFeedInfoFeedLangMatch.java
- tests: https://github.com/MobilityData/gtfs-validator/blob/v1.4.0/usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateAgencyLangAndFeedInfoFeedLangMatchTest.java

### PR DESCRIPTION
**Pull request by lionel-nj**
closes #524

**Summary:**

This PR implements a new validator: to validate validate `feed_info.feed_lang` and `agency.agency_lang` match

**Expected behavior:** 
- [x] no notice should be generated if `feed_info.txt` is empty
- [x] no notice should be generated if `feed_info.feed_lang` and `agency.agency_lang` match
- [x] notice should be generated if `feed_info.feed_lang` and `agency.agency_lang` do not match
- [x] notice should be generated if `feed_info.feed_lang` and the unique `agency.agency_lang` do not match
- [x] notice should be generated if `feed_info.feed_lang` is not "mul" and GTFS dataset contains more than one value for `agency.agency_lang`
- [x] notice should not be generated if `feed_info.feed_lang` is not "mul" and GTFS dataset contains only one matching value for `agency.agency_lang`
- [x] notice should generate notice if `feed_info.feed_lang` is "mul" and GTFS dataset contains no more than one value for `agency.agency_lang`
- [x] unit tests
  - [x] draft unit tests
  - [x] implement unit tests

Capture d’écran, le 2020-12-10 à 13 06 58

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- ~[ ] Format the title like "Fix #<issue_number> - <short description of fix and changes>" (for example - "Fix #1111 - Check for null value before using field")~
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: main/src/test/java/org/mobilitydata/gtfsvalidator/validator/MatchingFeedAndAgencyLangValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+        verify(mockNoticeContainer, times(1)).addNotice(captor.capture());
+
+        FeedInfoLangAndAgencyLangMismatchNotice notice = captor.getValue();
+        assertThat(notice.getCode()).matches("feed_info_lang_and_agency_lang_mismatch");
+        assertThat(notice.getContext()).containsEntry("feedInfoLang", "fra");
+        assertThat(notice.getContext()).containsEntry("agencyLangCollection", new HashSet<>(Arrays.asList("fra", "eng")));
+    }
+
+    @Test
+    public void feedLangNotMulAndOnlyThanOneMatchingAgencyLangShouldNotGenerateNotice() {
```
Comment (barbeau) [inline main/src/test/java/org/mobilitydata/gtfsvalidator/validator/MatchingFeedAndAgencyLangValidatorTest.java:126]: I think the `Than` should be removed from this method name?

File: main/src/test/java/org/mobilitydata/gtfsvalidator/validator/MatchingFeedAndAgencyLangValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+        FeedInfoLangAndAgencyLangMismatchNotice notice = captor.getValue();
+        assertThat(notice.getCode()).matches("feed_info_lang_and_agency_lang_mismatch");
+        assertThat(notice.getContext()).containsEntry("feedInfoLang", "fra");
+        assertThat(notice.getContext()).containsEntry("agencyLangCollection",
+                new HashSet<>(Arrays.asList("spa", "eng")));
+    }
+
+    @Test
+    public void mulFeedLandAnMoreThanOneAgencyShouldNotGenerateNotice() {
```
Comment (barbeau) [inline main/src/test/java/org/mobilitydata/gtfsvalidator/validator/MatchingFeedAndAgencyLangValidatorTest.java:259]: This method name has a typo too

**Review by barbeau — COMMENTED** [review]
Looks good @lionel-nj! Some minor comments in-line.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/MatchingFeedAndAgencyLangValidator.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    GtfsAgencyTableContainer agencyTable;
+
+    @Override
+    public void validate(NoticeContainer noticeContainer) {
+        if (feedInfoTable.isEmptyFile()) {
+            return;
+        }
+        // the previous early return ensures feedInfoTable.getEntities().stream() is not empty
+        @SuppressWarnings("OptionalGetWithoutIsPresent")
+        final Locale feedInfoFeedLang =
```
[the remark this answers is not in the document]
Comment (lionel-nj) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/MatchingFeedAndAgencyLangValidator.java:58]: Indeed, thanks

**Review by barbeau — APPROVED** [review]

**Pull request merged**